[Feat/#13] web bgcolor nav design

yh-project opened the pull request:
<!-- 풀 리퀘스트 제목
[<이슈 종류>/<이슈번호1>, <이슈번호2>] <제목>
-->

<!-- 리뷰어랑 담당자, 라벨 설정했는지 확인하세요 -->

## 🚀 Background

<!-- 어떤 걸 고치고 pr을 했는지 간단하게 -->
- 웹 사이트의 배경색을 설정
- 네비게이션 바 디자인
- 로고 이미지와 같은 리소스들을 저장하기 위한 폴더 추가

## 🥥 Contents

> 들어가기 전...

이번 프로젝트에서는 웹 사이트를 스타일링할 때에 TailwindCSS 를 이용해보기로 하였다. 따라서, 기존의 css 스타일링 방식과는 차이가 있을 수 있으며 TailwindCSS 에서 제공해주는 방식대로 className 속성에 모두 작성되고 공백으로 각각의 스타일이 구분된다.
<br>
### 웹 사이트 배경색 설정
유렌카의 사용자들이 사용하게될 웹 사이트의 가장 기본은 최상위 배경의 색을 지정하였으며 TailwindCSS 를 이용하기에 제공하는 방식으로 스타일이 작성되었으며 "slate/200" 으로 설정하였다. 웹 사이트 전반에 걸쳐 적용될 배경색이기 때문에 기본 페이지가 될 App.js 에 작성이되었다.
```javascript
// App.js
function App() {
  return (
    <>
      <div className="flex-col items-center justify-between w-screen h-screen bg-slate-200">
        <Nav name="홍길동" />
      </div>
    </>
  );
}
```
<br>

### 네비게이션 바 디자인
네비게이션 바는 웹 사이트 전반에 걸쳐 사용될 컴포넌트이기에 components 폴더 안에 추가하였으며, 크게 두 가지의 구역으로 나뉘게 된다.

첫 번째로는 로고 이미지를 띄우고 유렌카의 짧은 소개 멘트를 보여주게 된다. 로고 이미지의 경우에는 높이를 네비게이션 바에 맞추고 비율을 유지한채로 주어진 공간을 채우도록 object-contain 을 설정해주었다. 또한, 상하좌우에 패딩을 주었다.
```javascript
// Nav.js
// src 에 연결된 {logo} 의 경우 아래의 assets 폴더를 다룰 때 설명된다.
<img
       src={logo}
       alt="Logo"
       className="object-contain h-full px-2 py-3"
 ></img>
```
```CSS
/* object-contain */
object-fit : contain;
```
두 번째로는 사용자에게 제공될 기본 메뉴들과 사용자를 반겨줄 멘트를 보여주게 된다. 사용자를 반겨주는 멘트에는 사용자가 유렌카에 가입 시 입력한 닉네임을 삽입하여 보여주게 된다.
```javascript
// Nav.js
<li>
         <span className="text-amber-700">{props.name}</span> 고객님
         안녕하세요!
</li>
```
<br>

### assets 폴더 추가
로고 이미지와 같은 리소스들을 관리할 assets 폴더를 components, pages 폴더와 같은 위치에 추가해주었다. 네비게이션 바에 추가해줄 로고 이미지를 추가로 업로드하였다.
```
📦 src
├─ components
│  └─ Nav.js
├─ pages
├─ assets
│  └─ Logo.js
└─ App.js
```
©generated by [Project Tree Generator](https://woochanleee.github.io/project-tree-generator) <br>
아래와 같은 방식으로 사진을 import 하여 위에서 설명한 이미지 스타일링 코드의  src 속성에 연결해주게 된다.
```javascript
// Nav.js
import logo from "../assets/Logo.png";
```

<!--
코드, 개발 관점에서 어떤걸 고쳤는지 상세하게
사진같은걸 넣어도 된다.
pr 보는 사람이 따로 정보를 안 찾아봐도 되게 적는게 이상적
-->

## 🧪 Testing

- [x] 배경색이 제대로 적용되었는지 확인
- [x] 로고 이미지가 제대로 연결되어 보여지는지 확인
- [x] 사용자의 닉네임이 정상적으로 보여지는지 확인
- [x] 네비게이션 바가 의도한 대로 디자인되었는지 확인

<!-- 테스트 방법이나, 테스트 한 목록들을 적는다. -->

## 📸 Screenshot

<!-- 움짤을 넣어주는게 가장 좋고, 왠만하면 용량을 작게 만든다. -->
웹 사이트 전체 화면이며, 배경색이 제대로 적용되고 네비게이션 바가 정상적으로 상단에 위치하는 것을 확인할 수 있다.
![스크린샷 2023-05-21 오후 11 48 10](https://github.com/YU-RentCar/yurentcar-fe-web/assets/86611398/0de78bc7-7c78-4efd-9646-965b5f1f0c99)
<br>

로고 이미지가 정상적으로 출력된 것을 볼 수 있다.
![스크린샷 2023-05-21 오후 11 48 21](https://github.com/YU-RentCar/yurentcar-fe-web/assets/86611398/a483e29a-c1d4-4bdb-b68e-a75f1a46e58a)
<br>

임의로 지정한 사용자의 닉네임을 원하는 스타일로 정상적으로 출력된 것을 볼 수 있다.
![스크린샷 2023-05-21 오후 11 48 30](https://github.com/YU-RentCar/yurentcar-fe-web/assets/86611398/1a831ea1-f109-4c98-a7b3-d14f4413d48c)




## ⚓ Related Issue

<!-- 이슈 번호를 적어주면 되고, close 같은 자동 닫힘을 등록하여도 된다. -->
- #13

## 📚 Reference

<!-- 자신이 참조한 정보의 출처를 적는다. -->
- TailwindCSS documents
https://tailwindcss.com/docs/

seolbotdae committed:
PRMerge/#12: <develop -> main>

[PRMerge/#7, #9] README PRMerge 관련 내용 추가 및 기초 파일 구조 세팅 적용

yh-project committed:
Style/#13: 웹 사이트 배경색 및 네비게이션 바 디자인

웹 사이트 배경색 설정
 - slate/200 으로 설정

네비게이션 바 디자인
 - 상단에 fixed 로 디자인
 - 네비게이션 바 배경 sky/100 으로 설정

yh-project committed:
Setting/#13: 로고를 추가하기 위한 파일 구조 세팅

src 폴더 내에 assets 폴더 추가
 - 로고 이미지와 같은 리소스들을 보관하기 위함

네비게이션 바에서 로고 이미지 경로 설정
 - 네비게이션의 상단 좌측에 로고 이미지가 뜨도록 설정

seolbotdae commented on src/App.js:
```diff
@@ -1,24 +1,14 @@
-      </header>
-    </div>
+    <>
+      <div className="flex-col items-center justify-between w-screen h-screen bg-slate-200">
```
h-screen, w-screen 이 결려 있을 경우, 스크롤 할 만한 긴 요소가 생기면 제대로 반영되지 않는 상황이 생기는것 같음.
  
이 코드를 넣어보면 문제 상황을 확인할 수 있음.
```
<>
  <div className="flex items-center justify-between w-screen h-screen bg-slate-200">
    <Nav name="홍길동" />
    <div className="w-20 pt-20 bg-rose-500 h-[3000px]">dd</div>
  </div>
</>
```
- 좌우 스크롤이 생기는 문제가 있음.
- 스크롤 시 배경화면이 끊기는 문제가 있음.


```
<>
  <div className="flex items-center justify-between w-full h-full bg-slate-200">
    <Nav name="홍길동" />
    <div className="w-20 pt-20 bg-rose-500 h-[3000px]">dd</div>
  </div>
</>
```
- 둘다 full 로 변경해주면 문제가 해결됨

yh-project replied:
width 와 height 을 -full 로 수정하였습니다.
재확인 부탁드립니다.

seolbotdae reviewed (changes requested):
확인했습니다.
App.js 에서 일어날 수 있는 문제에 대해서만 수정해서 올려주실 수 있나요?

yh-project committed:
Style/#13: 웹 사이트 최상위 컨테이너 width height 수정

width height 지정 방식 수정
 - 스크롤이 필요한 요소에서 제대로 적용되지 않는 문제 발생
 - screen 사이즈가 아닌 100% 만큼으로 수정

seolbotdae reviewed (approved):
수정 확인 완료
